odl/solvers/functional/default_functionals.py

```diff
@@ -1029,6 +1028,68 @@ def convex_conj(self):
         return KullbackLeiblerCrossEntropy(self.domain, self.prior)
 
 
+class SeparableSum(Functional):
+
+    """The functional corresponding to separable sum of functionals.
+
+    The separable sum of functionals ``f_1, f_2, ..., f_n`` is given by::
+
+        h(x_1, x_2, ..., x_n) = sum_i^n f_i(x_i)
+
+    The separable sum is thus defined for any collection of functionals with
+    the same range.
```

Comment:
Write something about gradient, proximal and convex conjugate, which are the interesting properties here.